### PR TITLE
Add fn body for `process_eth1_data`

### PR DESCRIPTION
**Pull request by Cravtos**

### CODE REVIEW

File: src/phase0/state_transition/block_processing.rs [hunk 1]
```diff
@@ -236,16 +236,28 @@ fn process_eth1_data<
...
         MAX_PROPOSER_SLASHINGS,
         MAX_VALIDATORS_PER_COMMITTEE,
         MAX_ATTESTER_SLASHINGS,
         MAX_ATTESTATIONS,
         MAX_DEPOSITS,
         MAX_VOLUNTARY_EXITS,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
```
Comment (Cravtos) [inline src/phase0/state_transition/block_processing.rs:248]: Should I remove `Result<(), Error>` from the function return type if no error can be returned?

Comment (ralexstokes) [reply src/phase0/state_transition/block_processing.rs:248]: yes please!

File: src/phase0/state_transition/block_processing.rs [hunk 2]
```diff
@@ -236,16 +236,28 @@ fn process_eth1_data<
...
+    state.eth1_data_votes.push(body.eth1_data.clone());
+
+    let votes_count = state
+        .eth1_data_votes
+        .iter()
+        .filter(|&vote| *vote == body.eth1_data)
```
Comment (ralexstokes) [inline src/phase0/state_transition/block_processing.rs:254]: nitpick: 

i believe we can just do this
```suggestion
        .filter(|vote| *vote == body.eth1_data)
```

and there are sufficient blanket impls of `PartialEq` for refs that the compiler will do the right thing

and then we have less code to grok

**Review by ralexstokes — COMMENTED** [review]
looks great! i left one minor suggestion

we can imagine a more efficient implementation than this but for the time being let's just stick to the spec :)

File: src/phase0/state_transition/block_processing.rs (hunk 2 above)
Comment (Cravtos) [reply src/phase0/state_transition/block_processing.rs:254]: The issue is 'body.eth1_data' is not a ref. 

I could do
```suggestion
        .filter(|vote| **vote == body.eth1_data)
```
or
```suggestion
        .filter(|vote| *vote == &body.eth1_data)
```
but would that change anything?

File: src/phase0/state_transition/block_processing.rs (hunk 1 above)
Comment (Cravtos) [reply src/phase0/state_transition/block_processing.rs]: Done: da0a232

File: src/phase0/state_transition/block_processing.rs (hunk 2 above)
Comment (ralexstokes) [reply src/phase0/state_transition/block_processing.rs:329]: ah good point, i was confused about the `body` item!

**Review by ralexstokes — APPROVED** [review]
nice updates!

**Pull request merged**